pilot/pkg/model/service.go

```diff
@@ -257,7 +247,16 @@ func GetLocalityOrDefault(label, defaultLocality string) string {
 		// replace "." with "/"
 		return strings.Replace(label, k8sSeparator, "/", -1)
 	}
-	return defaultLocality
+	return defaultLabel
+}
+
+// Locality information for an IstioEndpoint
+type Locality struct {
+	// Label for locality on the endpoint. This is a "/" separated string.
```

Comment:
do we actually use the `/` separated form anywhere? Should we just parse this here instead of later in the code?

Reply:
But this is parsed, no? I think we just use this string directly throughout the xds code.

Reply:
Looks like there is just the envoy type: https://github.com/istio/istio/blob/6da78c79380ceeb795a8a1809406b6d4d09e1a56/pilot/pkg/networking/util/util.go#L299 and we use the string in some places. I think we used to have a similar struct internal but I guess we do not anymore, so this is probably fine